File: EngSoftTypeScript/src/VoarFoguete.ts
import {PadraoVoaveis} from "./PadraoVoaveis";

export class VoarFoguete implements PadraoVoaveis {

        private readonly velocidade: number;

        constructor() {
            this.velocidade = 1000;
        }

        voar(): string {
            return "Voando como um foguete. Velocidade: " + this.getVelocidade();
        }

        getVelocidade(): number {
            return this.velocidade;
        }

}

File: EngSoft.Jogo.Patos/src/engsoft/jogo/patos/Voar_Foguete.java
package engsoft.jogo.patos;

// Classe concreta
// Polimorfismo
public class Voar_Foguete implements Padrao_Voaveis {

    private final double velocidade;

    public Voar_Foguete() {
        velocidade = 1000;
    }

    public String voar() {
        return "Voando como um foguete. Velocidade: " + getVelocidade();
    }

    public double getVelocidade() {
        return velocidade;
    }

}
